hibernate-core/src/main/java/org/hibernate/query/spi/AbstractQuery.java

```diff
@@ -62,8 +63,15 @@
 import static org.hibernate.jpa.SpecHints.HINT_SPEC_QUERY_TIMEOUT;
 
 /**
+ * Base implementation of {@link org.hibernate.query.Query}.
+ *
+ * @apiNote This class is now considered internal implementation
+ * and will move to an internal package in a future version.
+ * Application programs should never depend directly on this class.
+ *
  * @author Steve Ebersole
  */
+@Internal
 public abstract class AbstractQuery<R>
```

Comment:
Hey @gavinking 

Should this really become an internal class? Hibernate Search extends it in a few places ... can we keep it an spi as it is now ?

Reply:
thanks 🙂